Source/WebKit/Shared/WebCoreArgumentCoders.cpp

```diff
@@ -1976,15 +1976,15 @@ template<class Encoder>
 void ArgumentCoder<PixelBuffer>::encode(Encoder& encoder, const PixelBuffer& pixelBuffer)
 {
     if (LIKELY(is<const ByteArrayPixelBuffer>(pixelBuffer))) {
-        downcast<const ByteArrayPixelBuffer>(pixelBuffer).encode(encoder);
+        encoder << downcast<ByteArrayPixelBuffer>(pixelBuffer);
         return;
     }
     ASSERT_NOT_REACHED();
 }
 
 std::optional<Ref<PixelBuffer>> ArgumentCoder<PixelBuffer>::decode(Decoder& decoder)
 {
-    return ByteArrayPixelBuffer::decode(decoder);
+    return decoder.decode<Ref<PixelBuffer>>();
```

Comment:
doesn't this cause infinite recursion?

Reply:
Maybe that’s why i have failing tests? :) I will investigate

Reply:
This was the issue. I had used to wrong type indeed :/